### PR TITLE
Add support for mutable updates. Follows Gl-matrix argument pattern

### PR DESCRIPTION
**Pull request by kevzettler**
Hey @chinedufn I was calling this in a loop and getting hit with Garbage Collector spikes. This is my attempt to make it support mutable updates following the gl-matrix argument pattern. Let me know your thoughts

### CODE REVIEW

**Comment by chinedufn** [comment]
@kevzettler nice thanks!! Mind just updating the test to verify that `out` ends up having the correct results?

Also looks like the existing [tests are currently failing in your branch](https://travis-ci.org/chinedufn/dual-quat-to-mat4/builds/286269959?utm_source=github_status&utm_medium=notification).

`npm run test` to verify locally

So yeah once tests are passing I think we're good to go!

---

I've added you to the library so feel free to merge and deploy a new major version to `npm` at your ready!

Thanks!

**Comment by chinedufn** [comment]
Oh sorry could you also please update the `Usage` and `API` sections of the README before merging?

Again, thanks!!

File: README.md [hunk 1]
```diff
@@ -28,19 +28,27 @@ $ npm install --save dual-quat-to-mat4
 ```js
 var dualQuatToMat4 = require('dual-quat-to-mat4')
 
-var dualQuat = dualQuatToMat4([0, 0, 0, 1, 0, 0, 0, 0])
+var dualQuat = dualQuatToMat4([], [0, 0, 0, 1, 0, 0, 0, 0])
```
Comment (chinedufn) [inline README.md]: Might just replace this line with:

```js
 var dualQuatToMat4 = require('dual-quat-to-mat4')

var matrix = new Float32Array(16)
dualQuatToMat4(matrix, [0, 0, 0, 1, 0, 0, 0, 0])

console.log(matrix)
// [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1]
```

In order to illustrate the recommended pre-allocated usage

File: README.md [hunk 2]
```diff
@@ -28,19 +28,27 @@ $ npm install --save dual-quat-to-mat4
...
 
-#### dual quaternion
+#### out
 
-*Requred*
+*Required* 
+
+Type: `Array[*]`
+
+An allocated array that will be mutated and ouput as the return value
```
Comment (chinedufn) [inline README.md]: Nice! 

Maybe:

> An allocated array that will get elements `[0]` through `[15]` set as your output matrix

?

File: README.md [hunk 3]
```diff
@@ -28,19 +28,27 @@ $ npm install --save dual-quat-to-mat4
...
-var dualQuat = dualQuatToMat4([0, 0, 0, 1, 0, 0, 0, 0])
+var dualQuat = dualQuatToMat4([], [0, 0, 0, 1, 0, 0, 0, 0])
 
 console.log(dualQuat)
 // [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1]
 ```
 
 ## API
 
-### `dualQuatToMat4(matrix)` -> `Array[16]`
```
Comment (chinedufn) [inline README.md:39]: ohhh nice catch

**Comment by kevzettler** [comment]
updated with documentation suggestions

**Comment by chinedufn** [comment]
Thanks!

**Pull request merged**